workflow/rules/common.smk

```diff
@@ -34,6 +34,20 @@ units = pd.read_csv(config["units"], sep="\t", dtype={"sample_name": str, "unit_
 validate(units, schema="../schemas/units.schema.yaml")
 
 
+def get_recalibrate_quality_bam_input(wildcards):
+    if is_activated("duplicate_filtering"):
+        return "results/dedup/{}.sorted.bam".format(wildcards.sample)
+    else:
+        return "results/mapped/{}.sorted.bam".format(wildcards.sample)
+
+
+def get_recalibrate_quality_bai_input(wildcards):
+    if is_activated("duplicate_filtering"):
+        return "results/dedup/{}.sorted.bam.bai".format(wildcards.sample)
+    else:
+        return "results/mapped/{}.sorted.bam.bai".format(wildcards.sample)
```

Comment:
Use only one function with a keyword argument `bai=False`. This way, it is not so duplicated.